contrib/opentimelineio_contrib/adapters/tests/test_aaf_adapter.py

```diff
@@ -884,6 +901,74 @@ def test_external_reference_paths(self):
                 self.assertEqual(clip.media_reference.target_url,
                                  audio_target_urls[track_index][clip_index])
 
+    def test_aaf_subclip_metadata(self):
+        """
+        For subclips, the AAF SourceClip can actually reference a CompositionMob
+        (instead of a MasterMob)
+        In which case we need to drill down into the CompositionMob
+        to find the MasterMob with the UserComments.
+        """
+
+        timeline = otio.adapters.read_from_file(SUBCLIP_PATH)
+        audio_track = timeline.audio_tracks()[0]
+        first_clip = audio_track[0]
+
+        aaf_metadata = first_clip.media_reference.metadata.get("AAF")
+
+        self._verify_user_comments(aaf_metadata)
+
+    def test_aaf_composition_metadata(self):
+        """
+        For standard clips the AAF SourceClip can actually reference a
+        CompositionMob (instead of a MasterMob) and the composition mob is holding the
+        UserComments instead of the MasterMob.
+        My guess is that the CompositionMob is used to share the same metadata
+        between different SourceClips
+        """
+
+        timeline = otio.adapters.read_from_file(COMPOSITION_METADATA_PATH)
+
+        audio_track = timeline.audio_tracks()[0]
+        first_clip = audio_track[0]
+
+        aaf_metadata = first_clip.media_reference.metadata.get("AAF")
+
+        self._verify_user_comments(aaf_metadata)
+
+    def test_aaf_composition_metadata_mastermob(self):
+        """
+        For standard clips the AAF SourceClip can actually reference a
+        CompositionMob (instead of a masterMob), the CompositionMob is holding
+        UserComments AND the MasterMob is holding UserComments.
+        In this case the masterMob has the valid UserComments (empirically determined)
+        """
+
+        timeline = otio.adapters.read_from_file(
+            COMPOSITION_METADATA_MASTERMOB_METADATA_PATH)
+
+        audio_track = timeline.audio_tracks()[0]
+        first_clip = audio_track[0]
+
+        aaf_metadata = first_clip.metadata.get("AAF")
+
+        self._verify_user_comments(aaf_metadata)
+
+    def _verify_user_comments(self, aaf_metadata):
+
+        self.assertTrue(aaf_metadata is not None)
+        self.assertTrue("UserComments" in aaf_metadata.keys())
+
+        user_comments = aaf_metadata['UserComments']
+
+        user_comment_keys = user_comments.keys()
+
+        # Check a few of the expected metadata keys of the test-AAF
+        self.assertTrue("Director" in user_comment_keys)
+        self.assertTrue("Line" in user_comment_keys)
+        self.assertTrue("Talent" in user_comment_keys)
+        self.assertTrue("Logger" in user_comment_keys)
+        self.assertTrue("Character" in user_comment_keys)
```

Comment:
It would be good to verify the values also.

Reply:
ok, will add value-validation

Reply:
Also, I think there is `assertIn`: https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertIn

Reply:
`self.assertIn(`

Reply:
It looks like `assertIn()` and `assertNotIn()` are only available in Python 3.1 and above, so we can't use them until we drop support for Python 2.7.